app/src/main/cpp/Pointer.cpp

```diff
@@ -171,9 +177,13 @@ Pointer::SetScale(const vrb::Vector& aHitPoint, const vrb::Matrix& aHeadTransfor
 
 void
 Pointer::SetPointerColor(const vrb::Color& aColor) {
+  if (aColor == m.pointerColor)
+      return;
+
   m.pointerColor = aColor;
   if (m.layer) {
     m.layer->SetTintColor(aColor);
+    m.layer->NotifySurfaceChanged(VRLayer::SurfaceChange::Create, NULL);
```

Comment:
This does not look incorrect, but I wonder why we didn't need it before. For example with the OculusVR backend we were already using layers and this was not required.

Reply:
Yeah, I agree it looks odd. Didn't find a better way to force a repaint of the layer as it is currently implemented.

> For example with the OculusVR backend we were already using layers and this was not required.

The SetPointerColor() hasn't been working with layers at least since we moved to OpenXR. The layer gets painted only once, after it is created, and never again. I'm not sure how it worked before, but I can take a look.

Reply:
Yes, please, do check why it was not needed with the OculusVR backend

Reply:
I looked how the oculusvr backend handles this and they have a special API in `ovrLayer` to change the tint color of the layer without requiring a repaint from Java side:

https://github.com/Igalia/wolvic/blob/acd800d40fc0a8f7a614e2fd8967b52fb387aa25/app/src/oculusvr/cpp/OculusVRLayers.h#L78  

This means that the `renderPointerLayer()` method in VRBrowserActivity was always been called once, at layer creation; but the OVR backend had the ability to change the layer *tint* color dynamically, and that's what was used to set the pointer color.

However with OpenXR we lost this functionality, so we need to find a way to either force a re-paint by calling `renderPointerLayer()` as I'm doing now by abusing the layer creation event; or re-factoring all the code to always paint the layer on cpp side, with vrb API. The latter option is the right one I think but will delay this change a bit, since doesn't appear trivial. WDYT?

Reply:
OK I understand, it's basically the same mechanism that was used to perform the background dimming when watching a fullscreen video, it's something that we need to figure out how to do in OpenXR, maybe with some shader. I'm fine with th e current solution.

Reply:
> That said, have you tried calling layer->RequestDraw() instead ? I know that it's only called from VRLayerNode currently but perhaps fits better than issuing a surface changed event.

Yes, but that doesn't work because note that we are calling RenderPointerLayer(), the Java painting code, in the lambda that runs when a `SurfaceChange` event arrives, and right now these events are only `Create` and `Destroy`.

What I can do is add a new event e.g, `Redraw` to SurfaceChange enum in VRLayer.h, and then call `NotifySurfaceChange()` as I do now but with a `Redraw` rather than `Create`. That will work and is more consistent than using `Create`.

WDYT?